### PR TITLE
Add white list for dist passes

### PR DESCRIPTION
**Pull request by sneaxiy**
### PR types
New features

### PR changes
Others

### Describe
Add white lists for better pass conflict processing.

### CODE REVIEW

**Review by aoyulong — DISMISSED** [review]
LGTM

File: python/paddle/distributed/passes/pass_base.py [hunk 1]
```diff
@@ -164,9 +166,61 @@ def _fusion_opt_last_rule(pass_before, pass_after):
...
+    for k, v in before_white_lists_dict.items():
+        for pass_name in v:
+            compatible_pass_dict[k].add(pass_name)
+
+    for k, v in after_white_lists_dict.items():
+        for pass_name in v:
+            compatible_pass_dict[pass_name].add(k)
+
+    def rule(pass_before, pass_after):
+        all_passes_before = compatible_pass_dict.get(pass_after.name)
```
Comment (zhiqiu) [inline python/paddle/distributed/passes/pass_base.py]: What is the meaning of  `all_passes_before`?

Comment (sneaxiy) [reply python/paddle/distributed/passes/pass_base.py]: All passes that can be applied after `pass_after`.

Comment (sneaxiy) [reply python/paddle/distributed/passes/pass_base.py]: Thanks for pointing out that I use the wrong variable name here. I would fix this in the next PR.

Comment (sneaxiy) [reply python/paddle/distributed/passes/pass_base.py]: Done.

**Review by zhiqiu — APPROVED** [review]
LGTM

**Pull request merged**